pkg/plugin/state.go

```diff
@@ -182,6 +182,16 @@ type podState struct {
 // podMigrationState tracks the information about an ongoing pod's migration
 type podMigrationState struct{}
 
+type milliCPU uint32
+
+func FromResourceQuantity(r resource.Quantity) milliCPU {
+	return milliCPU(r.MilliValue())
+}
+
+func (m *milliCPU) toResourceQuantity() resource.Quantity {
+	return *resource.NewMilliQuantity(int64(*m), resource.BinarySI)
+}
```

Comment:
This should be public, in `pkg/api`, and used there (and the autoscaler-agent) for CPU instead of `resource.Quantity`. This requires implementing custom formatting and JSON (un)marshalling, but I think that's worth the effort.

In particular, using `MilliCPU` in `api.Resources` should make a lot of the plugin changes simpler, I think.

Reply:
Realized the architecture doc was out of date, opened https://github.com/neondatabase/autoscaling/pull/179 which should hopefully make my reasoning for "this should be in `pkg/api` clearer"